share/python/driver.py

```diff
@@ -501,7 +1261,20 @@ def energy(name, **kwargs):
     >>> energy('fci', ref_wfn=cisd_wfn)
 
     """
+
+    if hasattr(name, '__call__'):
+        return name(energy, **kwargs)
+
     lowername = name.lower()
+
+    # Do a cp thing
+    if kwargs.get('bsse_type', None) is not None:
+        return _nbody_gufunc(energy, lowername, **kwargs)
+
+    # Check if this is a CBS extrapolation
+    if "/" in lowername:
+        return _cbs_gufunc('energy', lowername, **kwargs)
+
     kwargs = p4util.kwargs_lower(kwargs)
```

Comment:
Think you want the `kwargs_lower` absolutely at the top of the file so that `energy('mp2', BSSE_type='cp')` gets properly cased to be caught by `kwargs.get('bsse_type'`